src/main/java/io/tiledb/java/api/Util.java

```diff
@@ -37,6 +38,21 @@ public static int castLongToInt(long num) throws TileDBError {
     return (int) num;
   }
 
+  /**
+   * Checks if a given bigInteger value is in the UInt64 range
+   *
+   * @param bigInteger the bigInteger
+   * @return true if the range is ok
+   */
+  public static boolean checkBigIntegerRange(BigInteger bigInteger) throws TileDBError {
+    BigInteger upperLimit = new BigInteger("18446744073709551615");
```

Comment:
Probably not necessary right now, but if this ever ends up in a hot path we could encapsulate this in a helper class in order to make it static.